### ISSUE TITLE
2.1.0-rc2: Consul catalog provider does not check service health

### ISSUE THREAD

**Issue by byronwolfman** [containous/traefik#5905]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

The consul catalog provider automatically adds all service backends regardless of whether their healthcheck status is passing, warn, or critical. Ideally traefik should add/remove backends as they become healthy or unhealthy.

To test:

* Create a service definition with a healthcheck pointing to localhost:1234

```hcl
service {
  name = "broken"
  tags = [
    "traefik.enable=true",
  ]
  address = ""
  port = 1234
  token = "<redacted>"

  check = {
    id = "fail-check"
    name = "fail-check"
    tcp = "127.0.0.1:1234"
    interval = "10s"
    timeout = "1s"
  }
}
```

* Make sure nothing is actually listening on port 1234 so that the healthcheck fails

* Confirm the healthcheck is critical

```sh
$ curl -Ss \
    -H 'X-Consul-Token: <redacted>' \
    127.0.0.1:8500/v1/agent/health/service/name/broken | grep Status

        "AggregatedStatus": "critical",
                "Status": "critical"
```

### What did you expect to see?

Traefik should not route traffic to backends with critical healthchecks.

### What did you see instead?

Traefik dutifully routes traffic to backends with critical healthchecks.

```sh
$ curl localhost:80 -H 'Host: broken.dev.local' -X GET -I
HTTP/1.1 502 Bad Gateway
Date: Wed, 27 Nov 2019 16:04:46 GMT
Content-Length: 11
Content-Type: text/plain; charset=utf-8
```

![Screen Shot 2019-11-27 at 11 04 42 AM](https://user-images.githubusercontent.com/11906832/69739514-d8524380-1105-11ea-9d2e-b5d675e01b1d.png)

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      2.1.0-rc2
Codename:     cantal
Go version:   go1.13.4
Built:        2019-11-15T19:43:27Z
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```yaml
providers:
  providersThrottleDuration: '2s'

  consulCatalog:
    cache: True
    defaultRule: "Host(`{{ .Name }}.dev.local`)"
    endpoint:
      address: 'http://127.0.0.1:8500'
      token: '<redacted>'
    exposedByDefault: False
    requireConsistent: False
    stale: True
```

### If applicable, please paste the log output in DEBUG level (`--log.level=DEBUG` switch)

```
(paste your output here)
```

**Comment by SantoDE** [comment]
Hello @byronwolfman,

I just wanted to let you know quickly, that I'm able to reproduce your bug. I'm not really sure yet, where it actually comes from now how to fix it, but I'm on it.

Thanks for reporting.

**Comment by vvitayau** [comment]
Confirmed seeing this exists in traefik version 2.1.1,
using two vault servers where one is sealed and the other is unsealed.

**Comment by traefiker** [comment]
Closed by #6046.

**Comment by nicofuccella** [comment]
Pending the release...
